packages/compat/src/keypair.ts

```diff
@@ -0,0 +1,63 @@
+import { Keypair } from '@solana/web3.js';
+
+function addPkcs8Header(bytes: Uint8Array): Uint8Array {
+    // prettier-ignore
+    const header = new Uint8Array([
+        /**
+         * PKCS#8 header
+         */
+        0x30, // ASN.1 sequence tag
+        0x2e, // Length of sequence (46 more bytes)
+
+            0x02, // ASN.1 integer tag
+            0x01, // Length of integer
+                0x00, // Version number
+
+            0x30, // ASN.1 sequence tag
+            0x05, // Length of sequence
+                0x06, // ASN.1 object identifier tag
+                0x03, // Length of object identifier
+                    // Edwards curve algorithms identifier https://oid-rep.orange-labs.fr/get/1.3.101.112
+                        0x2b, // iso(1) / identified-organization(3) (The first node is multiplied by the decimal 40 and the result is added to the value of the second node)
+                        0x65, // thawte(101)
+                    // Ed25519 identifier
+                        0x70, // id-Ed25519(112)
+    ]);
+    const result = new Uint8Array(header.length + bytes.length);
+    result.set(header);
+    result.set(bytes, header.length);
+    return result;
+}
+
+/**
+ * Convert from a Legacy Web3 JS Keypair to a CryptoKeyPair
+ * @param keypair   The Keypair to convert
+ * @returns         A CryptoKeyPair
+ */
+export async function fromLegacyKeypair(keypair: Keypair): Promise<CryptoKeyPair> {
+    const publicKey: CryptoKey = await crypto.subtle.importKey(
+        'raw',
+        keypair.publicKey.toBytes(),
+        {
+            name: 'EdDSA',
+            namedCurve: 'Ed25519',
+        },
```

Comment:
This clashes with other code that uses the string `Ed25519` as the algorithm

https://github.com/solana-labs/solana-web3.js/blob/5e1b2e5d451f4016b1d846d84b3fa3309e43e494/packages/webcrypto-ed25519-polyfill/src/index.ts#L33

https://github.com/solana-labs/solana-web3.js/blob/5e1b2e5d451f4016b1d846d84b3fa3309e43e494/packages/addresses/src/public-key.ts#L7

https://github.com/solana-labs/solana-web3.js/blob/5e1b2e5d451f4016b1d846d84b3fa3309e43e494/packages/assertions/src/subtle-crypto.ts#L16

I'm not sure which is correct but I think we definitely need to consistently use the same algorithm in our code.

Reply:
It's the one you linked haha throws an error for me with what I had!